lib/puppet/defaults.rb

```diff
@@ -65,7 +65,11 @@ def self.default_vendormoduledir
 
   AS_DURATION = %q{This setting can be a time interval in seconds (30 or 30s), minutes (30m), hours (6h), days (2d), or years (5y).}
 
-  define_settings(:main,
+  # @api public
+  # @param args [Puppet::Settings] the settings object to define default settings for
+  # @return void
+  def self.initialize_default_settings!(settings)
```

Comment:
why !

Reply:
I used the `!` here because `initialize_default_settings!` modifies the settings object given to it.